### PR TITLE
Exclude certain weights from weight loading

### PR DESCRIPTION
**Pull request by LJFHendriks**

### CODE REVIEW

File: vedacore/misc/checkpoint.py [hunk 1]
```diff
@@ -281,6 +283,12 @@ def load_weights(model,
         state_dict = {k[7:]: v for k, v in state_dict.items()}
     if prefix is not None:
         state_dict = {'%s.%s' % (prefix, k): v for k, v in state_dict.items()}
+
+    if exclude is not None:
+        for k in list(state_dict.keys()):
+            if k.startswith(exclude):
+                del state_dict[k]
```
Comment (gauthsvenkat) [inline vedacore/misc/checkpoint.py:290]: Not too familiar/comfortable with the del keyword. @ReubenJ Can you also have a look?

**Comment by gauthsvenkat** [comment]
Nice Job! Seems like we can choose and load only the modules of the network we need.

File: vedacore/misc/checkpoint.py (hunk 1 above)
Comment (LJFHendriks) [reply vedacore/misc/checkpoint.py:290]: "del state_dict[k]" does exactly the same as "state_dict.pop(k)", so it removes the key-value pair with key "k" from the dictionary. Using del is just slightly more efficient than pop.

Comment (ReubenJ) [reply vedacore/misc/checkpoint.py:290]: Reads nicely as well :+1:

**Review by ReubenJ — APPROVED** [review]
Super nice solution :+1:

**Pull request merged**